packages/cms-cli/commands/ls.js

```diff
@@ -0,0 +1,131 @@
+const {
+  addPortalOptions,
+  addConfigOptions,
+  setLogLevel,
+  getPortalId,
+  addUseEnvironmentOptions,
+} = require('../lib/commonOpts');
+const { trackCommandUsage } = require('../lib/usageTracking');
+const { logDebugInfo } = require('../lib/debugInfo');
+const { validatePortal } = require('../lib/validation');
+const {
+  loadConfig,
+  validateConfig,
+  checkAndWarnGitInclusion,
+} = require('@hubspot/cms-lib');
+const { logger } = require('@hubspot/cms-lib/logger');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('@hubspot/cms-lib/errorHandlers');
+const {
+  getDirectoryContentsByPath,
+} = require('@hubspot/cms-lib/api/fileMapper');
+const {
+  HUBSPOT_FOLDER,
+  MARKETPLACE_FOLDER,
+} = require('@hubspot/cms-lib/lib/constants');
+
+const FOLDER_DOT_EXTENSIONS = ['functions'];
+
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath, options);
+  checkAndWarnGitInclusion();
+
+  if (!(validateConfig() && (await validatePortal(options)))) {
+    process.exit(1);
+  }
+};
+
+exports.command = 'ls [path]';
+exports.describe = 'get remote contents of a directory';
+
+exports.handler = async options => {
+  loadAndValidateOptions(options);
+
+  const { path } = options;
+  const directoryPath = path || '/';
+  const portalId = getPortalId(options);
+
+  trackCommandUsage('ls', {}, portalId);
+
+  logger.debug(`Getting contents of ${directoryPath}`);
+
+  const contentsResp = await getDirectoryContentsByPath(
+    portalId,
+    directoryPath
+  ).catch(async e => {
+    await logApiErrorInstance(
+      portalId,
+      e,
+      new ApiErrorContext({ portalId, directoryPath })
+    );
+    process.exit();
+  });
+
+  if (contentsResp.children.length) {
+    const mappedContents = contentsResp.children.map(fileOrFolder => {
+      const splitName = fileOrFolder.split('.');
+
+      if (
+        splitName.length > 1 &&
+        FOLDER_DOT_EXTENSIONS.indexOf(splitName[1]) === -1
+      ) {
+        return fileOrFolder;
+      }
+
+      return `/${fileOrFolder}`;
+    });
+    const hubspotFolder = `/${HUBSPOT_FOLDER}`;
+    const marketplaceFolder = `/${MARKETPLACE_FOLDER}`;
+
+    logger.log(
+      mappedContents
+        .sort(function(a, b) {
```

Comment:
I sort of wonder if we should use `String.localeCompare` to account for characters common to other languages.